src/spi.rs

```diff
@@ -779,6 +779,71 @@ macro_rules! hal {
                     Spi { spi, pins }
                 }
 
+                /// Enable transmit register empty interrupt
+                pub fn enable_txe_interrupt(&mut self) {
+                    self.spi.cr2.write(|w| { w.txeie().set_bit() });
```

Comment:
That does not look right, you'll have to use a read-modify-write operation, otherwise you're clearing all other flags.